src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,399 @@
+        guid = device.data["guid"]
+        name = device.data["name"]
+    if purge_date:
+        _update_cold_storage_purge_date(sdk, guid, purge_date)
+    if change_device_name and not name.startswith("deactivated_"):
+        _change_device_name(
+            sdk, guid, "deactivated_" + date.today().strftime("%Y-%m-%d") + "_" + name
+        )
+
+
+def _update_cold_storage_purge_date(sdk, guid, purge_date):
+    archives_response = sdk.archive.get_all_by_device_guid(guid)
+    archive_guid_list = [
+        archive["archiveGuid"]
+        for page in archives_response
+        for archive in page["archives"]
+    ]
+    for archive_guid in archive_guid_list:
+        sdk.archive.update_cold_storage_purge_date(archive_guid, purge_date)
+
+
+def _change_device_name(sdk, guid, name):
+    device_settings = sdk.devices.get_settings(guid)
+    device_settings.name = name
+    sdk.devices.update_settings(device_settings)
+
+
+_DEVICE_INFO_KEYS_MAP = OrderedDict()
+_DEVICE_INFO_KEYS_MAP["computerId"] = "Device ID"
+_DEVICE_INFO_KEYS_MAP["name"] = "Name"
+_DEVICE_INFO_KEYS_MAP["osHostname"] = "Hostname"
+_DEVICE_INFO_KEYS_MAP["guid"] = "GUID"
+_DEVICE_INFO_KEYS_MAP["status"] = "Status"
+_DEVICE_INFO_KEYS_MAP["archiveBytes"] = "Largest Archive Size in Bytes"
+_DEVICE_INFO_KEYS_MAP["lastConnected"] = "Last Connected Date"
+_DEVICE_INFO_KEYS_MAP["lastBackup"] = "Last Backup Date"
+_DEVICE_INFO_KEYS_MAP["lastCompletedBackup"] = "Last Completed Backup Date"
+_DEVICE_INFO_KEYS_MAP["productVersion"] = "Code42 Version"
+_DEVICE_INFO_KEYS_MAP["osName"] = "Operating System"
+_DEVICE_INFO_KEYS_MAP["osVersion"] = "Operating System Version"
+
+
+@devices.command()
+@device_id_option
+@format_option
+@sdk_options()
+def show(state, device_id, format=None):
+    """Print device info."""
+    formatter = OutputFormatter(format, _DEVICE_INFO_KEYS_MAP)
+    device_info = _get_device_info(state.sdk, device_id)
+    if device_info:
+        formatter.echo_formatted_list([device_info])
+
+
+def _get_device_info(sdk, device_id):
+    device = sdk.devices.get_by_id(device_id, include_backup_usage=True).data
+    device["archiveBytes"] = (
+        max(
+            [
+                backupDestination["archiveBytes"]
+                for backupDestination in device["backupUsage"]
+            ]
+        )
+        if len(device["backupUsage"]) > 0
+        else 0
+    )
+    device["lastBackup"] = (
+        max(
+            [
+                backupDestination["lastBackup"]
+                for backupDestination in device["backupUsage"]
+            ]
+        )
+        if len(device["backupUsage"]) > 0
+        else None
+    )
+    device["lastCompletedBackup"] = (
+        max(
+            [
+                backupDestination["lastCompletedBackup"]
+                for backupDestination in device["backupUsage"]
+            ]
+        )
+        if len(device["backupUsage"]) > 0
+        else None
+    )
+    return device
```

Comment:
A couple things here: 

1. As-is this is pretty hard to read and understand what's happening. Because we're checking `if len(device["backupUsage"]) > 0` in all 3 of these values, it would clarify things to break that out separately to make understanding how we're doing the calculations for values that do exist much easier:

```
    if not device["backupUsage"]:
        device["archiveBytes"] = 0
        device["lastBackup"] = None
        device["lastCompletedBackup"] = None
```
 
2. This currently fails if `backupDestination["lastCompletedBackup"]` is null/None as `max()` can't compare it. We'll need to filter out Nones, and then also handle an empty list.
3. Are we sure we want to reduce all these to single values across separate destinations? Imagine a device that has a really old 1TB archive on destination A that it hasn't backed up to in a year (for whatever reason), there is also destination B that was recently added, but the device has only backed up 1GB to it (maybe it's a different backup set with smaller selection), the device report would show this device as 1TB archive bytes with a very recent backup completed date.

Reply:
I'm updating it to handle null and empty cases more gracefully.

Regarding point 3, my goal with these processed values is to show a summary, the full detail (including the full backupUsage object) is available when using CSV or JSON formatting. 

When attempting to show information for all destinations, we run into trouble with the CSV format because the default output for a list is `[item1,item2,item3]`, which will behave oddly if they are all supposed to be in a single column (which they are, in this case). This could also present a problem for the CSV produced by `devices list` as well.

I think we'll need a helper method to format lists in a CSV-friendly manner (i.e. "item1 | item2 | item3" or something)

Reply:
Since the `show` command only handles one device at a time, I think it would make sense to show as much detail for it as possible. Could we just have each destination be it's own row so the user can get the full picture?

Reply:
I've updated show to output a second table with backup set information. This allows us to pull out this code entirely, since we no longer need to do any processing of the backup set information, and replace it with an output formatter.